### PR TITLE
Murisi/masp for namadillo

### PR DESCRIPTION
**Pull request by murisi**
## Describe your changes
Essentially https://github.com/anoma/namada/pull/3797, but with MASP hardware wallet integration disabled. Hardware wallet integration was disabled because (modified) ZIP 32  derivations are not yet supported by the device. Despite this, this PR is still useful because it separates MASP transaction building from MASP transaction signing, which is required for securely implementing MASP transactions on Namadillo.

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes
- [ ] If this PR requires changes to the docs or specs, a corresponding PR is opened in the `namada-docs` repo
    - Relevant PR if applies: 
- [ ] If this PR affects services such as `namada-indexer` or `namada-masp-indexer`, a corresponding PR is opened in that repo
    - Relevant PR if applies:

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -134,15 +134,15 @@ konst = { version = "0.3.8", default-features = false }
...
+ledger-namada-rs = { git = "https://github.com/Zondax/ledger-namada", tag = "v1.0.5" }
 ledger-transport = "0.10.0"
 ledger-transport-hid = "0.10.0"
 libc = "0.2.97"
 libloading = "0.7.2"
 linkme = "0.3.24"
 # branch = "tomas/arbitrary"
-masp_primitives = { git = "https://github.com/anoma/masp", rev = "12ed8b060b295c06502a2ff8468e4a941cb7cca4" }
-masp_proofs = { git = "https://github.com/anoma/masp", rev = "12ed8b060b295c06502a2ff8468e4a941cb7cca4", default-features = false, features = ["local-prover"] }
+masp_primitives = { git = "https://github.com/anoma/masp", rev = "2914e6ff9a922bae8f1cb63a79d796a69af3d8aa" }
```
Comment (tzemanovic) [inline Cargo.toml:144]: could you make a branch from this rev? I'm seeing `This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.` for it which might cause issues in build

File: crates/apps_lib/src/cli/wallet.rs [hunk 2]
```diff
@@ -184,13 +184,14 @@ fn payment_addresses_list(
 }
 
 /// Derives a masp spending key from the mnemonic code in the wallet.
-fn shielded_key_derive(
+async fn shielded_key_derive(
```
Comment (batconjurer) [inline crates/apps_lib/src/cli/wallet.rs:187]: why did this become async?

Comment (tzemanovic) [reply crates/apps_lib/src/cli/wallet.rs:187]: it's taken from #3797 without shielded keys support in HW wallet - we unfortunately already have some blocking calls mixed in the client already (e.g. wallet load) but I think we should just go with it and convert the blocking stuff to async

File: crates/apps_lib/Cargo.toml [hunk 3]
```diff
@@ -53,6 +53,7 @@ fd-lock.workspace = true
 flate2.workspace = true
 futures.workspace = true
 itertools.workspace = true
+jubjub.workspace = true
```
[the remark this answers is not in the document]
Comment (murisi) [reply crates/apps_lib/Cargo.toml]: Ah, this is a hangover from https://github.com/anoma/namada/pull/3797 where the client was doing jubjub computations. Removed. Thanks!

File: Cargo.toml (hunk 1 above)
Comment (murisi) [reply Cargo.toml]: Ah yes, I addressed your comment in the MASP crate and forgot to update the revision here. I've now merged `murisi/box-impl-build-params` into `main` in the MASP crate and changed the revision here to that. Thanks.

File: crates/core/src/masp.rs [hunk 4]
```diff
@@ -547,14 +549,22 @@ impl TransferSource {
...
-    pub fn spending_key(&self) -> Option<ExtendedSpendingKey> {
+    pub fn spending_key(&self) -> Option<PseudoExtendedKey> {
         match self {
-            Self::ExtendedSpendingKey(x) => Some(*x),
+            Self::ExtendedKey(x) => Some(*x),
+            _ => None,
+        }
+    }
+
+    /// Get the contained ExtendedSpendingKey contained, if any
```
[the remark this answers is not in the document]
Comment (murisi) [reply crates/core/src/masp.rs]: The comment was inaccurate. Now resolved.

**Review by mateuszjasiuk — COMMENTED** [review]
Everything works great from perspective of frontend integration :)!

**Pull request merged**